app/controllers/placements_controller.rb

```diff
@@ -44,7 +44,9 @@ def update
     @placement = Placement.find(params[:id])
 
     @placement.update!(placement_params)
-    redirect_to session.delete(:return_to)
+    if session[:return_to] != nil
+      redirect_to session.delete(:return_to)
+    end
```

Comment:
Here you should have an else, to redirect to the homepage if session[:return_to] is nil.  I missed that when we were working together on this.